tests/test_notebooks.py

```diff
@@ -0,0 +1,82 @@
+import glob
+import os
+import traceback
+
+import pytest
+from testbook import testbook
+
+import composer
+
+NB_PATH = '../notebooks/up_and_running_with_composer.ipynb'
+nb_root = os.path.join(os.path.dirname(composer.__file__), '..', 'notebooks')
+
+NOTEBOOKS = [
+    os.path.join(nb_root, nb) \
+    for nb in glob.glob(os.path.join(nb_root, '*.ipynb')) \
+]
+
+
+def seek(tb, tag):
+    """ generator that seeks next cell with the metadata tag,
+    and returns the cell index.
+    """
+    for idx, cell in enumerate(tb.cells):
+        metadata = cell['metadata']
+        if 'tags' in metadata and tag in metadata['tags']:
+            yield idx
+
+
+@pytest.fixture(params=NOTEBOOKS)
+def tb(request):
+    notebook_path = request.param
+    with testbook(notebook_path) as tb:
+        yield tb
+
+
+def _assert_tags_correct(tb):
+    """Checks that the trainer_fit tag exists
+       whenever trainer.fit is called.
+    """
+    for cell in tb.cells:
+        if cell['cell_type'] == 'code' and 'trainer.fit' in cell['source']:
+            metadata = cell['metadata']
+            if 'tags' not in metadata or 'trainer_fit' not in metadata['tags']:
+                raise ValueError("In cell that starts with '{}', trainer.fit was found "
+                                 "but the trainer_fit tag was not in metadata.".format(cell['source'][:40]))
+
+
+@pytest.mark.timeout(120)
```

Comment:
Can we shorten the notebook timeout tests to a max of 15 seconds? 2 minutes is kinda long...